libnmstate/nm/connection.py

```diff
@@ -164,3 +164,25 @@ def _logging_connection_info(con_setting, source):
         con_setting.props.autoconnect,
         con_setting.props.autoconnect_slaves
     )
+
+
+def get_ip4_profile(active_connection):
```

Comment:
I think that routes can depend on the ipv4 module.
I think it will be cleaner if the connection module will not know about the setting details, passing that knowledge to the other modules. Otherwise you will get into this module all setting getters from a lot of sources.